collections/associate_with.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Builds a new Record using the given array as keys and choosing a value for each
+ * key using the given selector.
+ *
```

Comment:
We should document what happens if the same key is processed multiple times (e.g. with `["a", "b", "a"]` input).